examples/chat/src/main.rs

```diff
@@ -108,13 +111,13 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // Kick it off
     loop {
         select! {
-            line = stdin.select_next_some() => {
+            Some(line) = stdin.next_line().then(|x| async move { x.expect("Stdin not to close") }) => {
```

Comment:
Can we make this simpler somehow? I'd like to avoid long chains here. Can't we just expand the pattern to `Some(Ok())`?

Reply:
I couldn't find a way to transpose() the Result/Option order (because of the Future) to keep the expect inside the block as before, so I've made this kind of ugly chain to keep the previous behavior of panicking if Err. 

But it can indeed a just a Ok(Some()), is the panicking desired behavior or simplicity is the most important thing here?

A third alternative would be transpose() inside the block, probably even worse:

            Some(line) = stdin.next_line().then(|x| async move { x.transpose() }) => {
                if let Err(e) = swarm
                    .behaviour_mut().gossipsub
                    .publish(topic.clone(), line.expect("Stdin not to close").as_bytes()) {
                    println!("Publish error: {e:?}");
                }
            }

Reply:
Well, this panic really should never happen, right? Hence the use of `expect` in the first place :)

Reply:
> simplicity is the most important thing here?

I'd argue yes, we want these examples to be simple!

Reply:
okay, made the changes :)